### PR TITLE
Add Save/Load Button Functionality

### PR DESCRIPTION
**Pull request by FinleyMcIlwaine**

### CODE REVIEW

File: src/Editor.tsx [hunk 1]
```diff
@@ -49,7 +58,34 @@ export class Editor extends React.Component<{
...
+              const changeHandler = () => {
+                const reader = new FileReader();
+                reader.readAsText(inputFile.files[0]);
+                reader.addEventListener(
+                  "load",
+                  () => {
+                    this.props.onChange(reader.result);
+                    inputFile.removeEventListener("change", changeHandler);
+                  },
+                  false
```
Comment (calebegg) [inline src/Editor.tsx]: I think the third parameter defaults to 'false', so you can omit it.

File: src/Editor.tsx [hunk 2]
```diff
@@ -49,7 +58,34 @@ export class Editor extends React.Component<{
               <path d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z" />
             </svg>
           </button>
-          <button aria-label="Load">
+          <input
+            type="file"
```
Comment (calebegg) [inline src/Editor.tsx]: sorry, I think I preferred the `ref=` thing here. `ref` makes it more obvious we're doing something kinda fishy. also makes it easier to write unit tests.

File: src/Editor.tsx [hunk 3]
```diff
@@ -49,7 +58,34 @@ export class Editor extends React.Component<{
...
+            id="fileInput"
+            accept=".lisp"
+            style={{ display: "none" }}
+          />
+          <button
+            onClick={() => {
+              const input = document.getElementById("fileInput");
```
Comment (calebegg) [inline src/Editor.tsx]: I'm still not sure why we need both `input` and `inputFile` (below). Aren't they the same?

Comment (FinleyMcIlwaine) [reply src/Editor.tsx]: You're right, they are. I was wrong in thinking the query selector was necessary as well.

**Comment by FinleyMcIlwaine** [comment]
Put the `ref=` back in, removed the `false`, and got rid of the unnecessary `input` variable. Let me know if I should make any other changes!

**Comment by calebegg** [comment]
Awesome, merged and pushed to new.proofpad.org. Thanks so much for the contribution!

I did make a few changes. There were some TypeScript compilation errors that I fixed. In the process, I found that you weren't using `this.fileInput` but instead `fileInput`. This worked only because you also had `id="fileInput"` which creates an implicit global variable `fileInput`. A bit scary, since if this component had been used multiple times on the page the references would have been broken. JavaScript is full of bad old behavior like this, which is one of the reasons I really like TypeScript :)

**Pull request merged**